src/caustic/lenses/utils.py

```diff
@@ -18,30 +18,23 @@ def get_pix_jacobian(
         Currently uses `jacrev` due to upcasting bug with `jacfwd` (https://github.com/pytorch/pytorch/issues/90065).
         Should probably switch to `jacfwd` in the future.
     """
-    jac = functorch.jacfwd(raytrace, (0, 1))(
-        thx, thy, z_l, z_s, cosmology, *args, **kwargs
-    )
+    jac = functorch.jacfwd(raytrace, (0, 1))(thx, thy, z_s, x)
```

Comment:
This is a good point, and also why I added an FFT method which is tried and tested on images 2000x2000 (I get magnification under a second on my laptop). I have not integrated it yet with base, but I suggest we take the same approach as with kappa grad, we give an option to toggle between the two methods and create a test to make sure they agree with each other.

Reply:
Thanks for adding this, @AlexandreAdam! One reason it's useful to have is because `get_magnification` uses `vmap` by default, which is sadly currently incompatible with `torch.autograd.Function`. There are some situations where this can be an issue (i.e. if the lens uses KeOps or `torchinterp1d`).

@ConnorStoneAstro: I'm wondering why is this going to be memory limited? To compute the magnification over an image, we `vmap` this:
https://github.com/Ciela-Institute/caustic/blob/b394cb6833323791a87d5b77c15b82f6090a4d60/src/caustic/lenses/utils.py#L37
I'm pretty sure one of the motivations for creating functorch was to efficiently compute batched Jacobians, so I'd expect this to work ok?

Reply:
I don't think it will be an issue for a while. Just jacobians in general are very memory intensive (Npix * Nparams) and then with automatic differentiation you also have to store the computational graph. In general the lowest memory way to do it is with finite differences, but that is a big pain. So for now I think this is the right way to go. I was just saying that we may need to get crafty in the future and break up the image into subsections and do the jacobian on one section at a time, or something more complicated like that.

Reply:
Here we're computing the Jacobian of `raytrace` (which outputs two coordinates) with respect to `thx` and `thy`, so the number of parameters isn't too large. But yes, we can revisit this if it's an issue. It looks like `vmap` has a `chunk_size` argument which could be very useful here!

I'll mark this as resolved and open an issue for @AlexandreAdam to write a test for the FFT method.